org-roam-link.el

```diff
@@ -0,0 +1,341 @@
+            (org-roam-link--auto-create-roam-link-file path manual)))))))
+
+(defun org-roam-link--current-buffer-roam-link-titles ()
+  "Return a list of unique roam-link titles in the current buffer."
+  (->> (org-element-map (org-element-parse-buffer) 'link
+         (lambda (link)
+           (let* ((type (org-element-property :type link))
+                  (path (org-element-property :path link))
+                  res)
+             (when (string= type "roam")
+               (cons path res)))))
+       (-flatten)
+       (-distinct)))
+
+(defun org-roam-link--completion (&optional _arg)
+  "Completion for roam-links in `org-mode'.
+ARG is optional prefix supplied through `org-mode'"
+  (let ((completions (org-roam--get-title-path-completions))
+        (in-buffer-completions (org-roam-link--current-buffer-roam-link-titles)))
+    (format "roam:%s" (completing-read "Roam note: "
+                                       (-union (map-keys completions) in-buffer-completions)))))
+
+(defun org-roam-link--status-message ()
+  "Send roam-link status message to minibuffer."
+  (when (and org-roam-verbose
+             (string= major-mode "org-mode"))
+    (let ((elem (org-element-context)))
+      (when (and (eq (car elem) 'link)
+                 (string= "roam" (org-element-property :type elem)))
+        (save-match-data
+          (let* ((title (org-element-property :path elem))
+                 (abs-file-path (org-roam--get-file-from-title title))
+                 (rel-file-path
+                  (when abs-file-path
+                    (string-match (format "%s\\(.*\\)" org-roam-directory) abs-file-path)))
+                 (file-path (when rel-file-path
+                              (concat "~org-roam-dir~" (match-string 1 abs-file-path))))
+                 (raw-link (org-element-property :raw-link elem)))
+            (if file-path
+                (org-roam-message "file: %s → %s" file-path raw-link)
+              (org-roam-message "No file found in db → %s" raw-link))))))))
+
+(defun org-roam-link-show-link-messages ()
+  "Enable minibuffer status message for roam-links.
+Follows example of `org-ref' and displays on idle timer."
+  (interactive)
+  (when org-roam-verbose
+    (or org-roam-link-message-timer
+        (setq org-roam-link-message-timer
+              (run-with-idle-timer 0.5 t #'org-roam-link--status-message)))))
+
+(defun org-roam-link-cancel-link-messages ()
+  "Disable minibuffer status message for roam-links."
+  (interactive)
+  (cancel-timer org-roam-link-message-timer)
+  (setq org-roam-link-message-timer nil))
+
+(defun org-roam-link-insert-link ()
+  "Shortcut to insert roam-link with standard completion prompt."
+  (interactive)
+  (insert "[[" (org-roam-link--completion) "]]"))
+
+(defun org-roam-link-insert-syntax ()
+  "Shortcut to insert roam-link syntax without completion.
+Move point inside brackets, ready for roam title entry."
+  (interactive)
+  (insert "[[roam:]]")
+  (backward-char 2))
```

Comment:
These two commands could stand to be combined into one (`org-roam-link-insert`) which changes behavior if called with a universal argument. The common case should be the default, the less common case should require the universal argument.

e.g. 
```emacs-lisp
(defun org-roam-link-insert (&optional completion)
    "Insert a :roam link.
If COMPLETION is non-nil,..."
; etc
)
```